File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.js
function solution(str_list) {
  if((!str_list.includes('l') && !str_list.includes('r')) || str_list.length === 1) return []

  if(!str_list.includes('r') && str_list.includes('l')) return str_list.slice(0, str_list.indexOf('l'));

  if(str_list.includes('r') && !str_list.includes('l')) return str_list.slice(str_list.indexOf('r') + 1)

  if(str_list.includes('l') && str_list.includes('r')){
    return str_list.indexOf('l') < str_list.indexOf('r') ? str_list.slice(0, str_list.indexOf('l')) : str_list.slice(str_list.indexOf('r') + 1)
  }
}

File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> solution(String[] str_list) {
        List<String> result = new ArrayList<>();

        int lIndex = -1;
        int rIndex = -1;
        
        for (int i = 0; i < str_list.length; i++) {
            if (str_list[i].equals("l") && lIndex == -1) {
                lIndex = i;
            }
            if (str_list[i].equals("r") && rIndex == -1) {
                rIndex = i;
            }
        }

        if (lIndex != -1 && (rIndex == -1 || lIndex < rIndex)) {
            for (int i = 0; i < lIndex; i++) {
                result.add(str_list[i]);
            }
        } else if (rIndex != -1) {
            for (int i = rIndex + 1; i < str_list.length; i++) {
                result.add(str_list[i]);
            }
        }
        return result;
    }
}